### DISCUSSION TITLE
Can't contact LDAP server

### DISCUSSION THREAD

**Discussion by SerhiiZahuba** [ltb-project/self-service-password#910] [Q&A]
Please help to find the problem. It is still not clear after what, but the server stopped connecting to ldap, I see an error in the logs

`2024/05/27 11:03:15 [error] 657#657: *32 FastCGI sent in stderr: "PHP message: PHP Warning:  ldap_bind(): Unable to bind to server: Can't contact LDAP server in /usr/share/self-service-password/htdocs/sendsms.php on line 128PHP message: LDAP - Bind error -1  (Can't contact LDAP server)PHP message: PHP Notice:  Undefined index: extended_error_msg in /var/cache/self-service-password/templates_c/efa37a1643088840655591baff46e9944b0ca30a_2.file.index.tpl.php on line 68PHP message: PHP Notice:  Trying to get property 'value' of non-object in /var/cache/self-service-password/templates_c/efa37a1643088840655591baff46e9944b0ca30a_2.file.index.tpl.php on line 68" while reading response header from upstream, client: 10.26.11.34, server: password-ua.domain.com, request: "POST / HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php7.4-fpm.sock:", host: "password`

my config

` LDAP
$ldap_url = "ldaps://ua.domain.com";
$ldap_starttls = false;
$ldap_binddn = "CN=service.login,OU=IT admins,OU=Services,DC=ua,DC=domain,DC=com";
$ldap_bindpw = 'kljhGHfd6675GGffgddsbn';
$ldap_base = "dc=ua,dc=domain,dc=com";
$ldap_login_attribute = "uid";
$ldap_fullname_attribute = "cn";
$ldap_filter = "(&(objectClass=user)(sAMAccountName={login})(!(userAccountControl:1.2.840.113556.1.4.803:=2)))";
$ldap_use_exop_passwd = false;
$ldap_use_ppolicy_control = false;
$ad_mode = true;`

**Comment by davidcoutadeur** [comment]
This is typically a connection problem. For debugging this issue, I suggest that you test your connection parameters with ldapsearch on the self-service-password machine:

```
ldapsearch -x -H 'ldaps://ua.domain.com' -D 'CN=service.login,OU=IT admins,OU=Services,DC=ua,DC=domain,DC=com' -W -b 'DC=ua,DC=domain,DC=com' -LLL -d -1
```

It can be a connectivity issue, a DNS issue, a certificate issue,...

**Reply by SerhiiZahuba** [reply]
Yes, the problem was in the certificate. Thank you for help.
